app/controllers/application_controller.rb

```diff
@@ -118,36 +118,21 @@ def acting_as?
     acting_user.object_id != session_user.object_id
   end
 
-  # Global exception handlers
-  rescue_from ActiveRecord::RecordNotFound do |exception|
-    Rails.logger.debug("#{exception.message}: #{exception.backtrace.join("\n")}") unless Rails.env.production?
-    render_404(exception)
-  end
-
-  rescue_from ActionController::RoutingError do |exception|
-    Rails.logger.debug("#{exception.message}: #{exception.backtrace.join("\n")}") unless Rails.env.production?
-    render_404(exception)
-  end
-
-  def render_404(_exception)
-    # Add html fallback in case the 404 is a PDF or XML so the view can be found
-    render "/404", status: 404, layout: "application", formats: formats_with_html_fallback
-  end
+  rescue_from NUCore::NotPermittedWhileActingAs, with: :render_acting_error
+  
+  # Test exception handlers
+  if Rails.env.test?
```

Comment:
Is there a way to test without this?  I usually try to avoid changing the application code to make testing work